setup/setuputils.class.inc.php

```diff
@@ -81,16 +81,23 @@ public static function CheckPhpAndExtensions()
 		$aResult = array();
 
 		// For log file(s)
-		if (!is_dir(APPROOT.'log'))
-		{
+		if (!is_dir(APPROOT.'log')) {
 			@mkdir(APPROOT.'log');
 		}
 
 		self::CheckPhpVersion($aResult);
 
 		// Check the common directories
-		$aWritableDirsErrors = self::CheckWritableDirs(array('log', 'env-production', 'env-production-build', 'conf', 'data'));
-		$aResult = array_merge($aResult, $aWritableDirsErrors);
+		$aWritableApprootDirsErrors = self::CheckWritableDirs(array('log', 'env-production', 'env-production-build', 'conf', 'data'));
+		$aResult = array_merge($aResult, $aWritableApprootDirsErrors);
+		// check temp dir (N°5235)
+		$sTmpDir = static::GetTmpDir();
+		clearstatcache(true, $sTmpDir);
+		if (is_writable($sTmpDir)) {
+			$aResult[] = new CheckResult(CheckResult::INFO, "The temp directory is writable by the application.");
+		} else {
+			$aResult[] = new CheckResult(CheckResult::ERROR, "The temp directory <b>'".$sTmpDir."'</b> is not writable for the application. Change its permission or use another dir (sys_temp_dir option in php.ini).");
```

Comment:
```suggestion
			$aResult[] = new CheckResult(CheckResult::ERROR, "The temp directory <b>'".$sTmpDir."'</b> is not writable by the application. Change its permission or use another dir (sys_temp_dir option in php.ini).");
```

To be consistent with the success message.